services/navigation/builtin/builtin_test.go

```diff
@@ -204,31 +350,75 @@ func TestStartWaypoint(t *testing.T) {
 			motionCfg *motion.MotionConfiguration,
 			extra map[string]interface{},
 		) (bool, error) {
-			err := kinematicBase.GoToInputs(ctx, referenceframe.FloatsToInputs([]float64{destination.Lat(), destination.Lng()}))
-			callChan <- struct{}{}
-			return true, err
+			callChan <- moveOnGlobeReq{
+				componentName:      componentName,
+				destination:        destination,
+				heading:            heading,
+				movementSensorName: movementSensorName,
+				obstacles:          obstacles,
+				motionCfg:          motionCfg,
+				extra:              extra,
+			}
+			return true, nil
 		}
-		pt := geo.NewPoint(1, 0)
-		err = ns.AddWaypoint(ctx, pt, nil)
+		ctx := context.Background()
+		logger := golog.NewTestLogger(t)
+		b := fakeBase(ctx, t, logger)
+		ns, teardown := setupNavService(ctx, t, injectMS, injectMSensor, b, logger)
+		defer teardown()
+
+		// add waypoint 1
+		pt1 := geo.NewPoint(1, 0)
+		err := ns.AddWaypoint(ctx, pt1, nil)
 		test.That(t, err, test.ShouldBeNil)
 
-		pt = geo.NewPoint(3, 1)
-		err = ns.AddWaypoint(ctx, pt, nil)
+		// test that added waypoints are returned by the waypoints function
+		wps, err := ns.Waypoints(ctx, nil)
 		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(wps), test.ShouldEqual, 1)
+		wpSimilar(t, wps[0], pointToWP(pt1))
 
-		ns.(*builtIn).mode = navigation.ModeManual
-		err = ns.SetMode(ctx, navigation.ModeWaypoint, nil)
+		// set mode to Waypoint to begin MoveOnMap calls
+		err = ns.SetMode(ctx, navigation.ModeWaypoint, map[string]interface{}{})
+		test.That(t, err, test.ShouldBeNil)
+
+		// confirm mode is now Waypoint
+		mode, err := ns.Mode(ctx, nil)
 		test.That(t, err, test.ShouldBeNil)
-		blockTillCallCount(t, 2, callChan, time.Second*5)
-		ns.(*builtIn).wholeServiceCancelFunc()
-		ns.(*builtIn).activeBackgroundWorkers.Wait()
+		test.That(t, mode, test.ShouldEqual, navigation.ModeWaypoint)
+
+		expectedMotionCfg := &motion.MotionConfiguration{
+			LinearMPerSec:     1,
+			AngularDegsPerSec: 1,
+		}
+
+		timeoutCtx, cancelFn := context.WithTimeout(context.Background(), time.Second)
+		defer cancelFn()
+
+		// confirm MoveOnGlobe is called for waypoint 1
+		expectedCall1 := moveOnGlobeReq{
+			componentName:      b.Name(),
+			destination:        pt1,
+			heading:            math.NaN(),
+			movementSensorName: injectMSensor.Name(),
+			motionCfg:          expectedMotionCfg,
+			// extra keeps default of "motion_profile": "position_only"
+			extra: map[string]interface{}{"motion_profile": "position_only"},
+		}
+		blockTillCalledOrTimeout(timeoutCtx, t, expectedCall1, callChan)
 
-		currentInputsShouldEqual(ctx, t, kinematicBase, pt)
+		// confirm waypoint 1 is eventually removed from the waypoints store after
+		// MoveOnGlobe reports reaching waypoint 1
+		wps1, err := pollTillWaypointLen(timeoutCtx, t, ns, 0)
```

Comment:
potentially can remove these from some of the tests if we're just testing things like the motion config

Reply:
done